### ISSUE TITLE
Incorrect flow stats for ng_netflow (FreeBSD)

### ISSUE THREAD

**Issue by shermdog** [netsampler/goflow2#186]
Hi Folks,

I've encountered an interesting bug / edge case with netflow v9 and ng_netflow from FreeBSD/OPNsense

The TL;DR is ng_netflow will always set `OUT_PKTS/OUT_BYTES` as `0` and includes them _after_ `IN_PKTS/IN_BYTES`

https://github.com/freebsd/freebsd-src/blob/release/13.2.0/sys/netgraph/netflow/netflow_v9.c#L278-L279
https://github.com/freebsd/freebsd-src/blob/release/13.2.0/sys/netgraph/netflow/netflow_v9.c#L78C1-L81

Unfortunately goflow2 has an implicit preference for OUT vs IN so I wind up with `0` byte/packet flows

https://github.com/netsampler/goflow2/blob/v1.3.4/producer/producer_nf.go#L289-L296

I believe the FreeBSD behaviour is suboptimal, but also _very_ pesimistic that a fix could be implemented there in a timely manner, so I'd like to see if you all have an opinon on how this could be handled in goflow2.

My initial thought is to ignore `0` `OUT_PKTS/OUT_BYTES` when `IN_PKTS/IN_BYTES` has data, but wanted to get your thoughts on a solution.

**Comment by shermdog** [comment]
Hi @lspgn - my particular use case makes it difficult to leverage custom mappings, as it's part of an agent.  I was able to fork and comment out the problematic bits for my particular build.  Happy to close this issue.

**Comment by lspgn** [comment]
Thank you @shermdog

You should be able to pass a `NetFlowMapper` to the `producer` if you're using it as a library. But a fork works too :) 

https://github.com/netsampler/goflow2/blob/9287e0552896c46181bd9e8204479e92a82de4b8/producer/producer_nf.go#L278

Feel free to reopen if you have more questions

**Comment by kellybyrd** [comment]
I just hit this bug using cloudflare/goflow so started looking at goflow2 and saw that it has the same problem. It seems valid for a NFv9 template to include both IN/OUT_BYTES (and PACKETS) but the FlowMessage protobuf only allows for a single `Bytes` value, so it's always going to throw something away. In what situations is OUT_BYTES used? Are there valid flows that have non-zero values for both? 

I can imagine the DIRECTION field would be useful, but the way parsing is done now in goflow2, you can't guarantee that DIRECTION has been parsed before IN/OUT_BYTES and because of NFv9 templates, DIRECTION may not be present at all.
